### PR TITLE
Adding group by, join queries

### PR DESCRIPTION
**Pull request by asafyos**

### CODE REVIEW

**Comment by NivBruker** [comment]
Commit without message

File: CoolBook/Controllers/AuthorsController.cs [hunk 1]
```diff
@@ -20,9 +21,54 @@ public AuthorsController(CoolBookContext context)
...
+                (author, book) => new
+                {
+                    Id = author.Id,
+                    Name = author.Name,
+                    BirthDate = author.BirthDate,
+                    Gender = author.Gender,
+                    Country = author.Country,
+                    BookId = book.Id
+                })
+                .GroupBy(x => new
```
Comment (NivBruker) [inline CoolBook/Controllers/AuthorsController.cs:54]: move the group by down and then you don't need the new?

File: CoolBook/Views/Users/Index.cshtml [hunk 2]
```diff
@@ -23,36 +23,59 @@
...
-                @Html.DisplayNameFor(model => model.UserName)
+                @Html.DisplayNameFor(model => model.User.UserName)
             </th>
             <th>
-                @Html.DisplayNameFor(model => model.Email)
+                @Html.DisplayNameFor(model => model.User.Email)
             </th>
             <th>
-                @Html.DisplayNameFor(model => model.Role)
+                @Html.DisplayNameFor(model => model.User.Role)
```
Comment (NivBruker) [inline CoolBook/Views/Users/Index.cshtml:32]: Why did you add these to the index?
The profile is where users can interact with this

File: CoolBook/Controllers/AuthorsController.cs (hunk 1 above)
Comment (asafyos) [reply CoolBook/Controllers/AuthorsController.cs:54]: this is how the group by works

File: CoolBook/Views/Users/Index.cshtml (hunk 2 above)
Comment (asafyos) [reply CoolBook/Views/Users/Index.cshtml:32]: In order to use JOIN, I needed a place without UserInfo, and considering the search option (which uses these parameters...)
